src/Models/UserModel.php

```diff
@@ -185,48 +184,49 @@ public function activate(User $user): void
     {
         $user->active = true;
 
-        $return = $this->save($user);
-
-        $this->checkQueryReturn($return);
+        $this->saveWithEmailIdentity($user);
     }
 
     /**
-     * Override the BaseModel's `save()` method to allow
-     * updating of user email, password, or password_hash
-     * fields if they've been modified.
-     *
-     * @param array|User $data
-     *
-     * @TODO can't change the return type to void.
+     * Save User and its Email Identity (email, password, or password_hash fields)
+     * if they've been modified.
      */
-    public function save($data): bool
+    public function saveWithEmailIdentity(User $data): void
     {
         try {
+            /** @throws DataException */
             $result = parent::save($data);
-
```

Comment:
I'm confused why we are moving this to the `catch` block? We would only end up there if there was no other data on the user to save, right? If the user has also been updated with username, or they have custom fields on the user table, like first/last names, the exception would never be thrown and the email/password would never be saved.

Reply:
The `$user->saveEmailIdentity()` in the catch block is called when there is no data to insert/update to the User table.
If `parent::save()` is successful, `$user->saveEmailIdentity()` will be called after the catch block.

Both `parent::save()` and `$user->saveEmailIdentity()` may throw DataException, and we can't distinguish them. 
Therefore I put only `parent::save()` in the try block.